File: Assignment #1/Codes/Task3.py
tupleList1 = [('Parent','Shahjahan','Arnob'),
		('Parent','Shahjahan','Anik'),
		('Parent','Shahjahan','Arnob'),
		('Parent','Shahjahan','Anna'),
		('Parent','Shahjahan','Ashim'),
		('Parent','Ashim','Manha')]

X = str(input("Grandchild: "))

print("Grandparent : ", end = ' ')
i = 0;
tupLen = len(tupleList1)
while(i < tupLen):
	if(tupleList1[i][0] == 'Parent')&(tupleList1[i][2] == X):
		par = tupleList1[i][1]
		for j in range(tupLen):
			if(tupleList1[j][0] == 'Parent') & (tupleList1[j][2] == par):
				grandpar = tupleList1[j][1]
				print(grandpar, end = ' ')
	i = i + 1;

File: Assignment #1/Codes/Task3.pl
parent('Shahjahan','Arnob').
parent('Shahjahan','Adit').
parent('Shahjahan','Anik').
parent('Shahjahan','Anna').
parent('Shahjahan','Amit').
parent('Shahjahan','Ashim').
parent('Ashim','Manha').



grandchild(X,Z):-
	parent(Z,Y),
	parent(Y,X).


findGp:-
	write('Grandkid: '),
	read(X),
	write('Grandparent: '),
	grandchild(X,Gp),
	write(Gp),
	tab(5),
	fail.